src/js/techreport/utils/data.js

```diff
@@ -100,6 +102,101 @@ const getLighthouseScoreCategories = (score, brackets) => {
+  const apis = [
+    {
+      endpoint: 'cwv',
+      metric: 'vitals',
+      parse: DataUtils.parseVitalsData,
+    },
+    {
+      endpoint: 'lighthouse',
+      metric: 'lighthouse',
+      parse: DataUtils.parseLighthouseData,
+    },
+    {
+      endpoint: 'adoption',
+      metric: 'adoption',
+      parse: DataUtils.parseAdoptionData,
+    },
+    {
+      endpoint: 'page-weight',
+      metric: 'pageWeight',
+      parse: DataUtils.parsePageWeightData,
+    },
+  ];
+
+  const pageNr = filters.page;
+  fetch(url)
+    .then(result => result.json())
+    .then(result => {
+      const category = result[0];
+      const firstTechNr = (pageNr - 1) * rows;
+      const lastTechNr = pageNr * rows;
+      const paginatedTechs = category?.technologies?.slice(firstTechNr, lastTechNr);
+
+      const technologyFormatted = paginatedTechs?.join('%2C')
+        .replaceAll(" ", "%20");
+
+      const compare = document.querySelector('[data-name="selected-apps"]');
+      compare.setAttribute('href', `/reports/techreport/tech?tech=${technologyFormatted}`);
+
+      const geo = filters.geo.replaceAll(" ", "%20");
+      const rank = filters.rank.replaceAll(" ", "%20");
+      const geoFormatted = geo.replaceAll(" ", "%20");
+      const rankFormatted = rank.replaceAll(" ", "%20");
+
+      let allResults = {};
+      paginatedTechs.forEach(tech => allResults[tech] = []);
+
+      Promise.all(apis.map(api => {
+        const url = `${Constants.apiBase}/${api.endpoint}?technology=${technologyFormatted}&geo=${geoFormatted}&rank=${rankFormatted}&start=latest`;
+
+        return fetch(url)
+          .then(techResult => techResult.json())
+          .then(techResult => {
+            techResult.forEach(row => {
+              const parsedRow = {
+                ...row,
+              }
+
+              if(api.parse) {
+                parsedRow[api.metric] = api.parse(parsedRow[api.metric], parsedRow?.date);
+              }
+
+              const resIndex = allResults[row.technology].findIndex(res => res.date === row.date);
+              if(resIndex > -1) {
+                allResults[row.technology][resIndex] = {
+                  ...allResults[row.technology][resIndex],
+                  ...parsedRow
+                }
+              } else {
+                allResults[row.technology].push(parsedRow);
+              }
+            });
+          });
+      })).then(() => {
+        category.data = {
+          technologies: allResults,
+          info: {
+            origins: category.origins,
+            technologies: category?.technologies?.length,
+          },
+        };
+
+        /* Update the pagination info */
+        const current = document.querySelectorAll('[data-page="current"]');
+        const total = document.querySelectorAll('[data-page="total"]');
+        current.forEach(c => c.innerHTML = pageNr);
+        total.forEach(t => t.innerHTML = Math.ceil(category?.technologies?.length / rows));
```

Comment:
```suggestion
        current.forEach(c => c.textContent = pageNr);
        total.forEach(t => t.textContent = Math.ceil(category?.technologies?.length / rows));
```

I seem obsessed now :-)